google-cloud-contrib/google-cloud-logging-adapters/cloud-logging-over-slf4j/src/main/java/org/slf4j/impl/CloudLogger.java

```diff
@@ -0,0 +1,172 @@
+      System.err.println("Error initializing cloud logging");
+      e.printStackTrace();
+    }
+  }
+
+  static void init() {
+    if (!INITIALIZED) {
+      String configFile = ConfigLoader.getValue(CLOUD_LOGGING_CONFIG, CONFIGURATION_FILE);
+      configLoader = new ConfigLoader(configFile);
+      logging = ILoggingFactory.get(configLoader.getDestination());
+      logging.init();
+      INITIALIZED = true;
+    }
+  }
+
+  String getName() {
+    return loggerName;
+  }
+
+  boolean isEnabled(Severity severity) {
+    return (isEnabled && severity.compareTo(minSeverity) >= 0);
+  }
+
+  void log(Marker marker, Severity severity, String text, Throwable throwable) {
+    writeLogEntries(marker, severity, text, throwable);
+  }
+
+  void log(Severity severity, String text, Throwable throwable) {
+    writeLogEntries(null, severity, text, throwable);
+  }
+
+  void log(Marker marker, Severity severity, String text) {
+    writeLogEntries(marker, severity, text, null);
+  }
+
+  void log(Severity severity, String text) {
+    log(null, severity, text, null);
+  }
+
+  private void updateConfig() {
+    minSeverity = configLoader.getSeverity(loggerName);
+    labels = configLoader.getLabels(loggerName);
+    fileName = configLoader.getFileName(loggerName);
+    this.resource = MonitoredResource.newBuilder(configLoader.getResource(loggerName)).build();
+    this.isEnabled = configLoader.isEnabled(loggerName);
+  }
+
+  private void init(String loggerName) throws Exception {
+    this.loggerName = loggerName;
+    updateConfig();
+  }
+
+  private LogEntry createLogEntry(Marker marker, Severity severity, String text) {
+    LogDetailsTuple logDetailsTuple = new LogDetailsTuple(marker, labels, severity);
+    return LogEntry.newBuilder(StringPayload.of(text))
+        .setResource(resource)
+        .setLabels(logDetailsTuple.getLabels())
+        .setLogName(fileName)
+        .setSeverity(logDetailsTuple.getLevel())
+        .build();
+  }
+
+  private List<String> getChunks(String text) {
+    List<String> chunks = new ArrayList<>();
+    if (text.length() <= MAX_LOG_SIZE_IN_CHARS) {
+      chunks.add(text);
+      return chunks;
+    }
+    String[] splits = text.split("\n");
+    StringBuilder sb = new StringBuilder();
+    for (String s : splits) {
+      if (s.length() + sb.length() < MAX_LOG_SIZE_IN_CHARS) {
+        sb.append("\n");
+        sb.append(s);
+      } else {
+        if (sb.length() > 0) {
+          chunks.add(sb.toString());
+          sb.setLength(0);
+        }
+        if (s.length() <= MAX_LOG_SIZE_IN_CHARS) {
+          sb.append(s);
+        } else {
+          int start = 0;
+          int end;
+          while (start < s.length()) {
+            end = Math.min(s.length() - start, MAX_LOG_SIZE_IN_CHARS - 1);
+            chunks.add(s.substring(start, start + end));
+            start = end;
+          }
+        }
+      }
+    }
+    if (sb.length() > 0) {
+      chunks.add(sb.toString());
+    }
```

Comment:
This was done to better handle segmenting around new lines, but i realize it looks too complicated. I've switched to using Splitter.fixedLength